### PR TITLE
Fix Logging HttpRequest#method

### LINKED ISSUE #1798: Google::Cloud::Logging::Entry overrides ruby reserved word
https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-logging/lib/google/cloud/logging/entry/http_request.rb#L36

vs

```
$ irb
2.4.1 :001 > class Kakka
2.4.1 :002?>   def kekke
2.4.1 :003?>     end
2.4.1 :004?>   end
 => :kekke
2.4.1 :005 > k = Kakka.new
 => #<Kakka:0x007f84ae08acc0>
2.4.1 :006 > k.method
ArgumentError: wrong number of arguments (given 0, expected 1)
	from (irb):6:in `method'
	from (irb):6
	from /Users/mpa/.rvm/rubies/ruby-2.4.1/bin/irb:11:in `<main>'
2.4.1 :007 > k.method :kekke
 => #<Method: Kakka#kekke>
```

### PR DESCRIPTION
**Pull request by blowmage**
`HttpRequest#method` is stomping on `Object#method` and this is preventing calls to `Object#method` to be made. This fix is to maintain both behaviors by checking whether an argument was passed to `#method`. The `HttpRequest#method` behavior is used when called with no arguments, and the `Object#method`, behavior is used when called with arguments.

```ruby
require "google/cloud/logging"

logging = Google::Cloud::Logging.new

entry = logging.entry
entry.http_request.method = :GET

# call with HttpRequest#method behavior
entry.http_request.method #=>:GET

# call with Object#method behavior
entry.http_request.method(:validated)
#=> #<Method: Google::Cloud::Logging::Entry::HttpRequest#validated>
```

[fixes #1798]

### CODE REVIEW

**Comment by quartzmo** [comment]
@blowmage What do you think about @dazuma's [suggestion](https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/1798#issuecomment-343310109) to rename `method` to `request_method`?

> I'd advocate for renaming the accessor to request_method since that's the name of the underlying proto field anyway.

**Comment by blowmage** [comment]
I think that would be a breaking change and I'd prefer not to force a 2.0 version to fix this. Thankfully the Logging `HttpRequest#method` and the `Object#method` signatures are different, so we can support both.

I am not opposed to renaming the method when we do move to 2.0 though.

**Comment by quartzmo** [comment]
Not a breaking change: Move docs/recommended usage to `request_method`, but keep support for `method` as you have done here. I think it's a good idea so that users don't feel worried or concerned about the overload of `method`.

**Comment by blowmage** [comment]
K, I added `HttpRequest#request_method` and added deprecation notice to the `HttpRequest#method` documentation. Calls to `HttpRequest#method` will continue to work so existing code won't need to be changed.

**Comment by blowmage** [comment]
@dazuma I'd really like your approval before accepting this.

File: google-cloud-logging/lib/google/cloud/logging/entry/http_request.rb [hunk 1]
```diff
@@ -33,7 +33,30 @@ def initialize
...
+
+          ##
+          # @overload method()
+          #   Deprecated. Use {#request_method} instead.
+          #
+          #   The request method. Examples: `"GET"`, `"HEAD"`, `"PUT"`,
+          #   `"POST"`. (String)
+          def method *args
+            # Call Object#method when args are present.
+            return super if args.any?
```
Comment (dazuma) [inline google-cloud-logging/lib/google/cloud/logging/entry/http_request.rb]: Should this be `unless args.empty?`?
because `[nil].any? # => false`

Comment (blowmage) [reply google-cloud-logging/lib/google/cloud/logging/entry/http_request.rb]: @dazuma Updated with test coverage.

**Pull request merged**